packages/cli/src/run.js

```diff
@@ -1,31 +1,23 @@
 import consola from 'consola'
 import NuxtCommand from './command'
-import * as commands from './commands'
-import setup from './setup'
 import listCommands from './list'
+import setup from './setup'
 
 export default function run() {
-  const defaultCommand = 'dev'
-  let cmd = process.argv[2]
-
-  if (commands[cmd]) { // eslint-disable-line import/namespace
-    process.argv.splice(2, 1)
-  } else {
+  const cmd = process.argv[2] || 'dev'
+  try {
+    NuxtCommand.ensure(cmd)
+  } catch (notFoundError) {
     if (process.argv.includes('--help') || process.argv.includes('-h')) {
-      listCommands().then(() => process.exit(0))
-      return
+      return listCommands().then(process.exit)
```

Comment:
@pi0 this is because process.exit() runs with status 0 by default :)

Reply:
I think this should be `then(() => process.exit(0)).catch(() => process.exit(1)` (Currently it is evaluated in parallel to listCommands)

Reply:
@pi0 A reference is passed to then(), there's no immediate call there :) The difference is in providing an existing function and passing a wrapper to an existing function.

Reply:
Good call about adding `catch()` though!

Reply:
Oops, you are right. But I suggest for code clarity avoid direct passing reference.